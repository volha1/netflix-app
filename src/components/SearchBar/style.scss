@import '../../common/scss/base.scss';

.search-bar {
  width: 90%;
  margin: 0 auto;

  .title {
    font-size: 40px;
    font-weight: 300;
    text-transform: uppercase;
    margin-bottom: 40px;
  }

  .search-field {
    box-sizing: border-box;
    width: 700px;
    height: 57px;
    padding: 0px 10px;
    margin-right: 10px;
    background-color: rgba(32, 32, 32, 0.8);
  }

  ::placeholder {
    color: $white;
    opacity: 0.3;
  }

  .search-btn {
    background-color: $red;
    width: 200px;
    height: 57px;

    &:hover {
      background-color: $dark-red;
    }
  }
}
