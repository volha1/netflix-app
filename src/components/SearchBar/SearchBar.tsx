/* eslint-disable arrow-body-style */
/* eslint-disable @typescript-eslint/no-explicit-any */
import React, { ReactElement, Dispatch, SetStateAction, useCallback, useState } from 'react';
import SearchParams from '../../types/SearchParams';
import './style.scss';

type SearchBarProps = {
  onSearch: Dispatch<SetStateAction<SearchParams>>;
  params: SearchParams;
  removeSearchParams: Dispatch<SetStateAction<string>>;
};

const SearchBar = ({ params, onSearch, removeSearchParams }: SearchBarProps): ReactElement => {
  const [search, setSearch] = useState(params.search);

  const handleChange = useCallback(
    (event: any) => {
      if (!event.target.value.trim()) {
        removeSearchParams('search');
        removeSearchParams('searchBy');
      }
      setSearch(event.target.value);
    },
    [search]
  );

  const handleClick = useCallback(() => {
    onSearch({ ...params, search, searchBy: 'title' });
  }, [params, search]);

  return (
    <div className="search-bar">
      <h1 className="title">Find your movie</h1>
      <input className="search-field" placeholder="What do you want to watch?" value={search} onChange={handleChange} />
      <button type="button" className="search-btn" onClick={handleClick}>
        Search
      </button>
    </div>
  );
};

export default SearchBar;
