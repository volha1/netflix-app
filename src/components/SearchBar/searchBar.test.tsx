import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

import SearchBar from './SearchBar';

describe('Search bar test', () => {
  const onSearch = jest.fn();
  const removeSearchParams = jest.fn();
  const params = { filter: '', sortOrder: '', sortBy: '', search: '', movie: '' };

  it('Search bar snapshot', () => {
    const component = renderer.create(
      <SearchBar params={params} onSearch={onSearch} removeSearchParams={removeSearchParams} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Title is present', () => {
    const { getByText } = render(
      <SearchBar params={params} onSearch={onSearch} removeSearchParams={removeSearchParams} />
    );

    expect(getByText('Find your movie')).toBeInTheDocument();
  });

  it('onSearch function works', async () => {
    const textForSearch = 'React';
    render(<SearchBar params={params} onSearch={onSearch} removeSearchParams={removeSearchParams} />);

    await userEvent.type(screen.getByRole('textbox'), textForSearch);
    await userEvent.click(screen.getByRole('button'));

    expect(onSearch).toHaveBeenCalled();
    expect(onSearch).toHaveBeenCalledWith({ search: textForSearch, searchBy: 'title' });
  });

  it('removeSearchParams function works', async () => {
    const textForSearch = ' ';
    render(<SearchBar params={params} onSearch={onSearch} removeSearchParams={removeSearchParams} />);

    await userEvent.type(screen.getByRole('textbox'), textForSearch);
    await userEvent.click(screen.getByRole('button'));

    expect(removeSearchParams).toHaveBeenCalledWith('search');
    expect(removeSearchParams).toHaveBeenCalledWith('searchBy');
  });
});
