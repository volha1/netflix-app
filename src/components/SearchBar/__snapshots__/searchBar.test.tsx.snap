// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar test Search bar snapshot 1`] = `
<div
  className="search-bar"
>
  <h1
    className="title"
  >
    Find your movie
  </h1>
  <input
    className="search-field"
    onChange={[Function]}
    placeholder="What do you want to watch?"
    value=""
  />
  <button
    className="search-btn"
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</div>
`;
